api/python/provisioner/commands/check.py

```diff
@@ -220,13 +266,55 @@ def get_passed(self) -> list:
         """
         return [check for check in self._check_entries if check.is_passed]
 
-    def get_failed(self) -> list:
+    def get_failed(self, critical: bool = None) -> list:
         """
-        Return all check entries which failed
-        Returns: list with all failed checks
+        Return all check entries which failed. If critical is set to True,
+        return only critical checks that failed. If critical is set to False,
+        return only non-critical checks that failed. If critical is set to
+        None, return all failed checks.
+
+        :param bool critical: parameter to filter returning check results
 
+        :return: list with failed checks satisfying the given condition
         """
-        return [check for check in self._check_entries if check.is_failed]
+        if critical is None:
```

Comment:
I think we can keep the previous version but instead of `self._check_entries` use result of `get_checks`

Reply:
I agree, it will take a less lines of code. Done